### PR TITLE
Fx device id test with OpenMPTarget

### PR DESCRIPTION
**Pull request by dalg24**
Has been failing since the merge of #6713

### CODE REVIEW

**Comment by dalg24** [comment]
Retest this please

File: core/unit_test/UnitTest_DeviceAndThreads.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ int get_device_id() {
 #elif defined(KOKKOS_ENABLE_HIP)
   KOKKOS_IMPL_HIP_SAFE_CALL(hipGetDevice(&device_id));
 #elif defined(KOKKOS_ENABLE_OPENMPTARGET)
-  device_id   = omp_get_device_num();
+  device_id   = omp_get_default_device();
 #elif defined(KOKKOS_ENABLE_OPENACC)
   device_id   = acc_get_device_num(acc_get_device_type());
```
Comment (crtrott) [inline core/unit_test/UnitTest_DeviceAndThreads.cpp:50]: isn't this wrong too??

**Comment by dalg24** [comment]
Retest this please

File: core/unit_test/UnitTest_DeviceAndThreads.cpp (hunk 1 above)
Comment (dalg24) [reply core/unit_test/UnitTest_DeviceAndThreads.cpp:50]: How so?

**Comment by dalg24** [comment]
HIP failure was clearly unrelated.  The OpenMPTarget build is passing.

**Review by masterleinad — APPROVED** [review]
Looks correct to me.

**Pull request merged**